### PR TITLE
Use a macro to create and update fdw server and user mappings

### PR DESCRIPTION
**Pull request by marcospri**
Copied the macros from the report project.

The goal is to both:

- Settle in one way to do this across multiple projects.
- Allow to add other mappings in a more concise manner.

For now this change should not have any effect, it should render the same SQL statements.

### CODE REVIEW

File: lms/data_tasks/report/refresh/00_fdw_refresh.jinja2.sql [hunk 1]
```diff
@@ -1,11 +1,17 @@
-ALTER SERVER "h_server" OPTIONS(
-    SET host '{{h_fdw.host}}', -- SECRET
-    SET port '{{h_fdw.port}}', 
-    SET dbname '{{h_fdw.dbname}}'
-);
+{% macro refresh_fdw_server(server_name, credentials, users) %}
```
Comment (marcospri) [inline lms/data_tasks/report/refresh/00_fdw_refresh.jinja2.sql:1]: Copied/Adapted from: https://github.com/hypothesis/report/blob/main/report/data_tasks/report/refresh/00_schema/01_fdw_refresh.jinja2.sql

File: lms/data_tasks/report/create_from_scratch/00_schema/02_create_fdw_server.jinja2.sql [hunk 2]
```diff
@@ -3,18 +3,27 @@
...
-);
 
-DROP USER MAPPING IF EXISTS FOR "{{db_user}}" SERVER "h_server";
-CREATE USER MAPPING IF NOT EXISTS FOR "{{db_user}}"
-    SERVER "h_server"
-    OPTIONS (
-        user '{{h_fdw.user}}',
-        password '{{h_fdw.password}}' -- SECRET
-);
+{% macro create_fdw_server(server_name, credentials, users) %}
```
Comment (marcospri) [inline lms/data_tasks/report/create_from_scratch/00_schema/02_create_fdw_server.jinja2.sql:7]: From https://github.com/hypothesis/report/blob/main/report/data_tasks/report/create_from_scratch/00_schema/02_create_fdw_server.jinja2.sql#L6

**Review by indigobravo — APPROVED** [review]
Looks good to my eyes, Marcos. Can I suggest we keep an eye on it the first time it runs and be prepared to revert if needed.

**Comment by marcospri** [comment]
Thanks Ian

Sorry for the slow turnaround on this one. I'm closing this, it does have a refactor benefit but I don't think I'll use this soon. I'll reopen if we get there.

**Pull request closed without merging**